biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -2287,24 +2348,37 @@ subroutine DamageRecovery(csite,cpatch,ccohort,newly_recovered)
       ! To determine recovery, the first priority is to determine how much
       ! resources (C,N,P) are required to recover the plant to the target
       ! pool sizes of the next (less) damage class
-      
-      ! Target sapwood biomass according to allometry and trimming [kgC]
-      call bsap_allom(dbh,ipft, ccohort%crowndamage-1, canopy_trim,sapw_area,target_sapw_c)
-      ! Target total above ground biomass in woody/fibrous tissues  [kgC]
-      call bagw_allom(dbh,ipft, ccohort%crowndamage-1, target_agw_c)
-      ! Target total below ground biomass in woody/fibrous tissues [kgC] 
-      call bbgw_allom(dbh,ipft,target_bgw_c)
+
+      ! Target sapwood biomass according to allometry, trimming and phenology [kgC]
+      call bsap_allom(dbh,ipft, ccohort%crowndamage-1, canopy_trim, elongf_stem, &
+           sapw_area,target_sapw_c)
+      ! Target total above ground biomass in woody/fibrous tissues
+      ! according to allometry, trimming and phenology [kgC]
+      call bagw_allom(dbh,ipft, ccohort%crowndamage-1, elongf_stem, target_agw_c)
+      ! Target total below ground biomass in woody/fibrous tissues
+      ! according to allometry, trimming and phenology [kgC]
+      call bbgw_allom(dbh,ipft, elongf_stem, target_bgw_c)
       ! Target total dead (structrual) biomass [kgC]
       call bdead_allom( target_agw_c, target_bgw_c, target_sapw_c, ipft, target_struct_c)
-      ! Target fine-root biomass and deriv. according to allometry and trimming [kgC, kgC/cm]
-      call bfineroot(dbh,ipft,canopy_trim,ccohort%l2fr,target_fnrt_c)
-      ! Target storage carbon [kgC,kgC/cm]
+      ! Target fine-root biomass according to allometry, trimming and phenology [kgC]
+      call bfineroot(dbh,ipft,canopy_trim,ccohort%l2fr, elongf_fnrt, target_fnrt_c)
+      ! Target storage carbon [kgC]
       call bstore_allom(dbh,ipft,ccohort%crowndamage-1, canopy_trim,target_store_c)
-      ! Target leaf biomass according to allometry and trimming
-      if(ccohort%status_coh==leaves_on) then
-         call bleaf(dbh,ipft,ccohort%crowndamage-1, canopy_trim,target_leaf_c)
-      else
-         target_leaf_c = 0._r8
+      ! Target leaf biomass according to allometry, trimming and phenology [kgC]
+      call bleaf(dbh,ipft,ccohort%crowndamage-1, canopy_trim, elongf_leaf, target_leaf_c)
+
+
+      ! If plants are drought deciduous and are losing or lost all leaves, halt all
+      ! the growth.
+      if (is_hydecid_dormant) then
+         ! Drought-deciduous. Do not recover any tissue until the growing season
+         target_leaf_c   = 0._r8
+         target_fnrt_c   = 0._r8
+         target_sapw_c   = 0._r8
+         target_struct_c = 0._r8
+      elseif (is_sedecid_dormant) then
+         ! Cold-deciduous. Do not recover leaves until the growing season
+         target_leaf_c   = 0._r8
       end if
```

Comment:
Is the idea is that if plants are deciduous or semi-deciduous and  currently have no leaves or are in the process of  shedding they shouldn't allocate to crown recovery? I think that makes sense. But I wonder if we can check for this earlier where we  also check for whether  or not the cohort is damaged, and then return with a null pointer if so?

Reply:
I had added this to be consistent with what was already in place (which only looked into leaf biomass). But I agree that if all targets are zero, we could make an early return. Let me look into it.

Reply:
If semi-deciduous plants are not in the process of shedding but are also not fully flushed will they be able to recover crown damage? Or do they save carbon for future flushing?

Reply:
If semi-deciduous are not shedding leaves, then they are able to recover crown damage, even if they are not fully flushed. But they may not be in full recovery, because the target carbon will still be capped by the elongation factors.